zk_prover/src/circuits/merkle_sum_tree.rs

```diff
@@ -37,8 +34,10 @@ pub struct MstInclusionCircuit<const LEVELS: usize, const L: usize, const N_ASSE
     pub root_hash: Fp,
 }
 
-impl<const LEVELS: usize, const L: usize, const N_ASSETS: usize> CircuitExt<Fp>
-    for MstInclusionCircuit<LEVELS, L, N_ASSETS>
+impl<const LEVELS: usize, const N_ASSETS: usize> CircuitExt<Fp>
+    for MstInclusionCircuit<LEVELS, N_ASSETS>
+where
+    [(); 2 * (1 + N_ASSETS)]: Sized,
```

Comment:
Is there a specific reason for using the `unit` type here? 
If it's not necessary, could you please consider changing it to `usize` instead of `()` in every `where` clause?

Reply:
Good point! Fixed it in b089b09.

I also applied many fixes to `backend` when merging and resolving conflicts from master in 1eef622. Can you have a look to that?

I think we should merge #116 first and then we can merge this!